### PR TITLE
GameDB: HW fixes for Mega Man/Rockman X7 + Clamping mode fix

### PR DESCRIPTION
**Pull request by Immersion95**
-SoftwareRendererFMVHack for all versions of Mega Man/Rockman X7
-Clamping mode fix for camera issues across all versions
-Same compat info across all versions

### CODE REVIEW

**Comment by Tzerinas** [comment]
Fixes should be after the compatiblity info for consistency.

File: bin/resources/GameIndex.yaml [hunk 1]
```diff
@@ -20509,6 +20511,11 @@ SLKA-25112:
 SLKA-25115:
   name: "Rockman X7"
   region: "NTSC-K"
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
+  compat: 5
```
Comment (Tzerinas) [inline bin/resources/GameIndex.yaml:20516]:
```suggestion
  compat: 5
  gameFixes:
    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
```

Compatiblity info should come before the settings themselves.

File: bin/resources/GameIndex.yaml [hunk 2]
```diff
@@ -21275,6 +21282,11 @@ SLPM-60172:
 SLPM-60207:
   name: "Rockman X7"
   region: "NTSC-J"
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
+  compat: 5
```
Comment (Tzerinas) [inline bin/resources/GameIndex.yaml:21287]:
```suggestion
  compat: 5
  gameFixes:
    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
```

Ditto

File: bin/resources/GameIndex.yaml [hunk 3]
```diff
@@ -24435,8 +24447,11 @@ SLPM-65330:
   name: "Akudaikan 2"
   region: "NTSC-J"
 SLPM-65331:
-  name: "RockMan X7"
+  name: "Rockman X7"
   region: "NTSC-J"
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
+  compat: 5
```
Comment (Tzerinas) [inline bin/resources/GameIndex.yaml:24454]:
```suggestion
  compat: 5
  gameFixes:
    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
```

Ditto

File: bin/resources/GameIndex.yaml [hunk 4]
```diff
@@ -11860,8 +11860,10 @@ SLES-51883:
   name: "Scooby-Doo! Mystery Mayhem"
   region: "PAL-M3"
 SLES-51885:
-  name: "MegaMan X7"
+  name: "Mega Man X7"
   region: "PAL-M5"
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
   compat: 5
```
Comment (Tzerinas) [inline bin/resources/GameIndex.yaml:11867]:
```suggestion
  compat: 5
  gameFixes:
    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
```

Comp info should be before the settings themselves.

File: bin/resources/GameIndex.yaml [hunk 5]
```diff
@@ -36995,8 +37010,10 @@ SLUS-20486:
   region: "NTSC-U"
   compat: 5
 SLUS-20487:
-  name: "MegaMan X7"
+  name: "Mega Man X7"
   region: "NTSC-U"
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
   compat: 5
```
Comment (Tzerinas) [inline bin/resources/GameIndex.yaml:37017]:
```suggestion
  compat: 5
  gameFixes:
    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
```

Ditto

**Comment by Immersion95** [comment]
Is it correct now ?

File: bin/resources/GameIndex.yaml [hunk 6]
```diff
@@ -20509,6 +20511,11 @@ SLKA-25112:
 SLKA-25115:
   name: "Rockman X7"
   region: "NTSC-K"
+  compat: 5
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
+  clampModes:
+    eeClampMode: 3  # For camera issues in some scenes.
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:20518]: remove extra space before #

File: bin/resources/GameIndex.yaml [hunk 7]
```diff
@@ -21275,6 +21282,11 @@ SLPM-60172:
 SLPM-60207:
   name: "Rockman X7"
   region: "NTSC-J"
+  compat: 5
+  gameFixes:
+    - SoftwareRendererFMVHack # Flickering and wrong textures in FMV.
+  clampModes:
+    eeClampMode: 3  # For camera issues in some scenes.
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:21289]: remove extra space before #

**Comment by refractionpcsx2** [comment]
just fix those two things, I'll deal with squashing. I recommend you get a decent Git client and do some research in how to rebase, squash and force push.

We don't really do merge commits here, they get messy, so we usually rebase and force push.  But, don't worry about it this time, I'll squash it when I merge.

**Comment by Immersion95** [comment]
Done. Sorry for my noobness :p.

**Pull request merged**